server/config/database.yml

```diff
@@ -1,25 +1,18 @@
-# SQLite version 3.x
-#   gem install sqlite3
-#
-#   Ensure the SQLite 3 gem is defined in your Gemfile
-#   gem 'sqlite3'
-#
+
 default: &default
-  adapter: sqlite3
-  pool: <%= ENV.fetch('RAILS_MAX_THREADS') { 5 } %>
+  adapter: postgresql
+  encoding: unicode
+  username: <%= ENV['POSTGRES_USER'] %>
```

Comment:
Another option is to use ENV['DATABASE_URL'] instead. Then you would only need to set one value (and you could keep supporting sqlite, making it easier to onboard new people (to coding/rails) without requiring them to have  postgresdb running

Reply:
Hi 
I think it is a good idea to make both SQLite and PostgreSQL available. 
I can set up Postgres database using a URL alone (although would this mean passing the password through the URL, is that a good idea?). 
 But I can't set up the SQLite databases with a URL right? I seem to need to provide the DB paths to save to. If I pass in a blank URL then I get an error.

Reply:
**Example:** This refers to the SQLite database file located at the local path `db/development.sqlite3`:

`DATABASE_URL=sqlite://db/development.sqlite3`

Reply:
I don't think I understand where the URL should go. 

At the moment I am imagining you make the changes in the .env file, with same database.yml for both scenarios. 
 At the moment the yml looks like this, but it doesn't work with the URL field included for SQL (haven't tried POSTGRES yet)
default: &default
  adapter: <%= ENV['ADAPTER'] %>
  encoding: unicode
  url:  <%= ENV['DATABASE_URL'] %>
  pool: 5
  timeout: 5000

development:
  <<: *default
  database: <%= ENV['DB'] %>
test:
  <<: *default
  database: <%= ENV['DB_TEST'] %>
production:
  <<: *default
  database: <%= ENV['DB'] %>

Reply:
With the yaml file  above  if I set the env parameters as below 
ADAPTER='sqlite3'
DB_TEST='db/mentormentee_test.sqlite3'
DB='db/mentormentee.sqlite3'
DATABASE_URL='sqlite3://..'
the databases are created but the rspec tests can't run due to: 
DatabaseCleaner::Safeguard::Error::RemoteDatabaseUrl:

Reply:
I am really struggling to combine the SQL and POSTGRES yaml files. 
Below works fine for SQL, but for POSTGRES ,passing a URL to DB and DB_TEST leads to the databases just being created by the superuser rather than a user which has been created for this project and specified in .env. 

Maybe the solution is to create two files (one for SQL and one for POSTGRES) and people can pick to keep one over the other? 

ADAPTER='sqlite3'
DB_TEST='sqlite3://../db/mentormentee_test.sqlite3'
DB='sqlite3://../db/mentormentee.sqlite3'

Reply:
Pushed another commit to show the issue between: 
Providing SQL with a DATABASE_URL seems to lead to problems creating an dev.sqlite3 and test.sqlite3 databases afterwards. 
Not providing POSTGRES with a DATABASE_URL and simply passing the URL at the point of creating the databases seems to mean that the superuser creates the database, not the user passed in the URL.